test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1

```diff
@@ -935,6 +935,172 @@ ConstructorTestClass(int i, bool b)
+                }
+                $attrCollection = [System.Collections.ObjectModel.Collection[System.Attribute]]::new()
+                $attrCollection.Add($attr)
+
+                $dynParam = [System.Management.Automation.RuntimeDefinedParameter]::new('Foo', [string], $attrCollection)
+
+                $paramDictionary = [System.Management.Automation.RuntimeDefinedParameterDictionary]::new()
+                $paramDictionary.Add('Foo', $dynParam)
+                $paramDictionary
+            }
+
+            end {}
+        }
+
+        $expected = '[string] Foo - Howdy partner'
+        $Script = 'Test-Function -'
+        $res = (TabExpansion2 -inputScript $Script).CompletionMatches | Where-Object CompletionText -eq '-Foo'
+        $res.Count | Should -Be 1
+        $res.CompletionText | Should -BeExactly '-Foo'
+        $res.ToolTip | Should -BeExactly $expected
+    }
+
+    It 'Should have type and name for parameter without help message' {
+        Function Test-Function {
+            param (
+                [Parameter()]
+                $WithParamAttribute,
+
+                $WithoutParamAttribute
+            )
+        }
+
+        $Script = 'Test-Function -'
+        $res = (TabExpansion2 -inputScript $Script).CompletionMatches |
+            Where-Object CompletionText -in '-WithParamAttribute', '-WithoutParamAttribute' |
+            Sort-Object CompletionText
+        $res.Count | Should -Be 2
+
+        $res.CompletionText[0] | Should -BeExactly '-WithoutParamAttribute'
+        $res.ToolTip[0] | Should -BeExactly '[Object] WithoutParamAttribute'
+
+        $res.CompletionText[1] | Should -BeExactly '-WithParamAttribute'
+        $res.ToolTip[1] | Should -BeExactly '[Object] WithParamAttribute'
+    }
+
+    It 'Should include help resource message in parameter tool tip by direct match' {
+        $expected = '`[string`] Activity - *'
+        $Script = 'Write-Progress -Activity'
+        $res = (TabExpansion2 -inputScript $Script).CompletionMatches
+        $res.Count | Should -Be 1
+        $res.CompletionText | Should -BeExactly '-Activity'
+        $res.ToolTip | Should -BeLikeExactly $expected
+    }
+
+    It 'Should include help resource message in parameter tool tip by multiple matches' {
+        $expected = '`[string`] Activity - *'
+        $Script = 'Write-Progress -'
+        $res = (TabExpansion2 -inputScript $Script).CompletionMatches | Where-Object CompletionText -eq '-Activity'
+        $res.Count | Should -Be 1
+        $res.CompletionText | Should -BeExactly '-Activity'
+        $res.ToolTip | Should -BeLikeExactly $expected
+    }
+    
+    It 'Should use first valid HelpMessage with multiple parameter attributes by direct match' {
+        Function Test-Function {
+            [CmdletBinding(DefaultParameterSetName = 'Default')]
+            param (
+                [Parameter(ParameterSetName = 'Foo', HelpMessage = 'Foo')]
+                [Parameter(ParameterSetName = 'Default')]
```

Comment:
I guess it should be:
```suggestion
                [Parameter(ParameterSetName = 'Foo', HelpMessage = 'Foo')]
                [Parameter(ParameterSetName = 'Default', HelpMessage = 'Second')]
```

Reply:
It's testing that the code will ignore an unset `HelpMessage` attribute and try the "next" one.

Reply:
I've updated the test name to hopefully make it clear what this is actually testing.